src/async-flat.mjs

```diff
@@ -1,31 +1,19 @@
 import { asyncIterableCurry } from './internal/async-iterable'
 
-const defaultShouldIFlat = (depth) => {
-  if (typeof depth === 'function') {
-    return depth
-  }
-  if (typeof depth === 'number') {
-    return (currentDepth, iter) =>
-      currentDepth <= depth &&
-      (typeof iter[Symbol.iterator] === 'function' || typeof iter[Symbol.asyncIterator] === 'function') &&
-      typeof iter !== 'string'
-  }
-  throw new Error('async-flat: "depth" can be a function or a number')
-}
+const defaultShouldFlat = item => (typeof item[Symbol.iterator] === 'function' || typeof item[Symbol.asyncIterator] === 'function') && typeof item !== 'string'
```

Comment:
you may want to split this function into multiple lines

Reply:
The reason I didn't is that once we are generating from templates we'll lose control of formatting, and the next changes to this file will likely be generating it from a template. Given that that's likely to be the next change to this file, I didn't want to cause unnecessary formatting churn.

If we want to maintain line-length-limited formatting for our generated code we'll need to adopt prettier, which is a whole other discussion.

Reply:
I see